node/mo/pages/api/links/index.ts

```diff
@@ -1,32 +1,20 @@
-import { NextApiRequest, NextApiResponse } from "next"
 import { connect } from "../../../utils/connection"
-import { ResponseFuncs } from "../../../utils/types"
+import { createMethodHandler } from "../../../utils/handle"
 
-const handler = async (req: NextApiRequest, res: NextApiResponse) => {
-    //capture request method, we type it as a key of ResponseFunc to reduce typing later
-    const method: keyof ResponseFuncs = req.method as keyof ResponseFuncs
-
-    //function for catch errors
-    const catcher = (error: Error) => res.status(400).json({ error })
-
-    // Potential Responses
-    const handleCase: ResponseFuncs = {
-        // RESPONSE FOR GET REQUESTS
-        GET: async (req: NextApiRequest, res: NextApiResponse) => {
-            const { MoLink } = await connect() // connect to database
-            res.json(await MoLink.find({}).sort({n: -1}).limit(100).catch(catcher))
-        },
-        // RESPONSE POST REQUESTS
-        POST: async (req: NextApiRequest, res: NextApiResponse) => {
-            const { MoLink } = await connect() // connect to database
-            res.json(await MoLink.create(req.body).catch(catcher))
-        },
-    }
-
-    // Check if there is a response for the particular method, if so invoke it, if not response with an error
-    const response = handleCase[method]
-    if (response) await response(req, res)
-    else res.status(400).json({ error: "No Response for This Request" })
-}
-
-export default handler
+export default createMethodHandler({
+  async get(req, resp) {
+    const { MoLink } = await connect();
+    resp.json(await MoLink.find().sort({ n: -1 }).limit(100));
```

Comment:
Isn't this missing the error handling now?

Reply:
I believe that errors here will be handled by NextJS as HTTP 500 errors, but I may be wrong.

Reply:
Could you confirm by making the sort invalid?

Reply:
Confirmed. It generates an HTTP 500 error report with a backtrace in the log and an error page in the UI.

Reply:
Thanks